filebeat/crawler/prospector.go

```diff
@@ -35,9 +36,42 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer = ProspectorStdin{
+			Prospector: p,
+		}
+	case cfg.LogInputType:
+		prospectorer = ProspectorLog{
+			Prospector: p,
+		}
+	}
+
+	if prospectorer == nil {
+		return fmt.Errorf("Invalid prospector type: %v", p.ProspectorConfig.Harvester.InputType)
+	}
+
+	p.prospectorer = prospectorer
+
 	return nil
 }
 
+// Starts scanning through all the file paths and fetch the related files. Start a harvester for each file
+func (p *Prospector) Run(spoolChan chan *input.FileEvent) {
+
+	p.running = true
+
+	logp.Info("Starting prospector of type: %v", p.ProspectorConfig.Harvester.InputType)
+
+	p.prospectorer.Run(spoolChan)
```

Comment:
we want to have the prospector provide a `Run` method or an `OnEvent` method to be called by loop in this method?